spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultStreamService.java

```diff
@@ -141,6 +141,24 @@ public void deployStream(String name, Map<String, String> deploymentProperties)
 		deployStreamWithDefinition(createStreamDefinitionForDeploy(name, deploymentProperties), deploymentProperties);
 	}
 
+	@Override
+	public void undeployStream(String streamName) {
+		StreamDeployment streamDeployment = this.streamDeploymentRepository.findOne(streamName);
+		if (streamDeployment != null) {
```

Comment:
Do we need to throw an Exception on the case where a Unknown Deployer Name is specified.

Reply:
Based on code I've seen in Skipper.  The answer is no.

Reply:
We have restrictions when storing the StreamDeployment entry as well.